lib/core-server/src/utils/open-in-browser.ts

```diff
@@ -1,15 +1,45 @@
 import { logger } from '@storybook/node-logger';
-import open from 'better-opn';
+import betterOpn from 'better-opn'; // betterOpn alias used because also loading open
+import open from 'open';
+import getDefaultBrowser from 'x-default-browser';
 import dedent from 'ts-dedent';
 
 export function openInBrowser(address: string) {
-  try {
-    open(address);
-  } catch (error) {
-    logger.error(dedent`
-      Could not open ${address} inside a browser. If you're running this command inside a
-      docker container or on a CI, you need to pass the '--ci' flag to prevent opening a
-      browser by default.
-    `);
-  }
+  const defaultBrowserCallback = async (err: any, res: any) => {
+    if (res.isFirefox) {
+      // Handle firefox cross-browser using open's system-specific binary utility.
+      // See https://github.com/sindresorhus/open#openapps for more information.
+      try {
+        await open(address, {
+          app: {
+            name: 'firefox',
+          },
+        });
+      } catch (error) {
+        logger.error(dedent`
+          Could not open ${address} inside a browser. If you're running this command inside a
+          docker container or on a CI, you need to pass the '--ci' flag to prevent opening a
+          browser by default.
+        `);
+      }
+    } else {
```

Comment:
shouldn't the logic be switched here where we use `betterOpn` for chrome/chromium only, and then use `open` for everything else? my understanding of the original issue is that `betterOpn` always prefers chrome if it's available and that's what was annoying users.

Reply:
This is also much simpler. I was originally turned off to this idea because of the way betterOpn triages versions of chrome. I updated to test for Chrome and checked it against every browser I could get my hands on for Mac and this approach works wonderfully, and it's much simpler to read. Thanks for the feedback and review!